addons/desec.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+
+# T&M Hansson IT AB © - 2020, https://www.hanssonit.se/
+
+true
+SCRIPT_NAME="deSEC Registration"
+SCRIPT_EXPLAINER="This script will automatically register a domain of your liking, and set it to automatically update your external IP address."
+# shellcheck source=lib.sh
+source /var/scripts/fetch_lib.sh || source <(curl -sL https://raw.githubusercontent.com/nextcloud/vm/master/lib.sh)
+
+# Maybe move to lib
+
+# Enter the subdomain
+msg_box "Please enter the subdomain (*example*.dedyn.io) that you want to use"
+while :
+do
+    SUBDEDYN=$(input_box_flow "Please enter the subdomain (*example*.dedyn.io) that you want to use \
+The only allowed characters for the username are:
+'a-z', 'A-Z', and '0-9'")
+    if [[ "$SUBDEDYN" == *" "* ]]
+    then
+        msg_box "Please don't use spaces."
+    elif [ "${SUBDEDYN//[A-Za-z0-9]}" ]
+    then
+        msg_box "Allowed characters for the username are:\na-z', 'A-Z', and '0-9'\n\nPlease try again."
+    else
+        DEDYNDOMAIN="$SUBDEDYN.dedyn.io"
+    fi
+    # Check for SOA record
+    if ! host -t SOA $DEDYNDOMAIN
+    then
+        msg_box "Sorry, but it seems like $DEDYNDOMAIN is taken. Please try with another domain."
+    else
+        break
+    fi
+done
+
+# Email address
+msg_box "You will now be prompted to enter an email address. It's very important that the email address you enter it a 100% valid one. deSEC will verify your email address by sending you a verification link.
+
+Every 6 months you will get an email asking you to confrim your domain. If deSeC doesn't get an answer within 3 weeks, you domain will be destroyed!"
+
+VALIDEMAIL=$(input_box_flow "Please enter a valid email address. NOT a fake or a temporary one.")
+
+msg_box "If you want to enter your account, please reset your password here:
+
+https://desec.io/reset-password"
```

Comment:
```suggestion
msg_box "```suggestion
msg_box "If you later want to log into your deSEC account, you need to set a login password here: https://desec.io/reset-password

You don't need to do this now."
```